### PR TITLE
Adds the functions min, max, sum, avg and mean

### PR DESCRIPTION
**Pull request by ChrSteinert**
I needed more functions for a Project using Math.NET. These are min, max, avg, mean and sum. With this PR I hope go get this into the official package and get some guidance on how to properly integrate this into Math.NET. The issues I am seeing right now:
1. `sum` is not a really pretty function. It was requested so I have it here for the time beeing. But I am happy to remove it again
2. The implementation for `median` just sits there in the match block that is otherwise tidy. I am missing a good spot to place it (and make sure it is correct ;) )
3. Function names in general. Maybe othernames, or `average` instead of `avg` should be preferred?
4. General formatting stuff.

I'd be happy to increment on this PR if there is interest in having those functions in Math.NET

Todos

- [ ] Missing simplifications (`min(1,2) ⇒ 1`)
- [ ] Missing tests
- [x] Missing implementation for applyN of Approximation.fs
- [x] Missing member for SymbolicExpression.fs
- [ ] Missing implementation for toLambda of Linq.fs
- [x] Missing implementation for functionName of Infix.fs
- [x] Missing implementation for functionName of VisualExpression.fs

### CODE REVIEW

**Comment by Happypig375** [comment]
`List.minBy`, `List.maxBy` and `List.averageBy` should be used.

File: src/Symbolics/Evaluate.fs [hunk 1]
```diff
@@ -226,6 +226,32 @@ module Evaluate =
...
+                    c 
+                    |> List.sort
+                    |> List.skip (len / 2 - 1)
+                    |> List.take 2
+                    |> median
+                | c when (c |> List.length) % 2 = 1 ->
+                    let len = c |> List.length
+                    c 
+                    |> List.sort
+                    |> List.skip (len / 2)
```
Comment (Happypig375) [inline src/Symbolics/Evaluate.fs]: Use `List.item`

**Comment by ChrSteinert** [comment]
Thanks – changed it.

File: src/Symbolics/Evaluate.fs [hunk 2]
```diff
@@ -226,6 +226,31 @@ module Evaluate =
...
+                | [c] -> c 
+                | [a; b] -> (a + b) / 2.
+                | c when (c |> List.length) % 2 = 0 ->
+                    let len = c |> List.length
+                    c 
+                    |> List.sort
+                    |> List.skip (len / 2 - 1)
+                    |> List.take 2
+                    |> median
+                | c when (c |> List.length) % 2 = 1 ->
```
[the remark this answers is not in the document]
Comment (Happypig375) [reply src/Symbolics/Evaluate.fs]: Tbh this should be placed in its own file - maybe called `Statistics.fs`, similar to `Calculus.fs`.

**Comment by Happypig375** [comment]
Cross-reference with https://github.com/mathnet/mathnet-symbolics/commit/deb77daf77e44ebe9f15b72376e6ce5f63272f5a:
- Missing tests
- Missing implementation for applyN of Approximation.fs
- Missing member for SymbolicExpression.fs
- Missing implementation for toLambda of Linq.fs
- Missing implementation for functionName of Infix.fs
- Missing implementation for functionName of VisualExpression.fs

File: src/Symbolics/Evaluate.fs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ChrSteinert) [reply src/Symbolics/Evaluate.fs]: Hm… The problem is, I cannot access the `FloatingPoint` type anymore from a different file, like `Calculus.fs`, which only operates on `Expression`s. I think that `mean` should operate on `FloatingPoint list`. What do you think?

**Comment by ChrSteinert** [comment]
I guess it makes sense to have `min`, `max` and Co also check for possible simplifications?
Like `min(1, 2) ⇒ 1`.
What do you think?

**Comment by FoggyFinder** [comment]
It would be awesome

**Comment by ChrSteinert** [comment]
Ok, then let's do this as well :)
Will add a todo.

**Pull request closed without merging**